codersdk/templates.go

```diff
@@ -229,6 +231,11 @@ type UpdateTemplateMeta struct {
 	// use the active version of the template. This option has no
 	// effect on template admins.
 	RequireActiveVersion bool `json:"require_active_version"`
+	// DeprecationMessage if set, will mark the template as deprecated and block
+	// any new workspaces from using this template.
+	// If passed an empty string, will remove the deprecated message, making
+	// the template usable for new workspaces again.
+	DeprecationMessage *string `json:"deprecation_message"`
```

Comment:
String or pointer to string? It's NOT NULL in the database, and exhaustruct should ensure that we _always_ fill out the fields.

Reply:
Pointer to string because empty string has meaning. If the user sends `""`, it removes the deprecation message.

This is on the `UpdateTemplateMeta`